conf/distro/include/fsl-base.inc

```diff
@@ -23,6 +23,13 @@ PREFERRED_PROVIDER_u-boot_mx7ulp = "u-boot-imx"
 PREFERRED_PROVIDER_u-boot_mx8mq = "u-boot-imx"
 PREFERRED_PROVIDER_u-boot_mx8mm = "u-boot-imx"
 
+# u-boot configuration in imx6qdlsabreauto, imx6qdlsabresd,
+# and imx6ulevk is set for fslc operation. Override it here
+# for fsl operation.
+UBOOT_SUFFIX_mx6 = "imx"
+SPL_BINARY_mx6 = ""
+UBOOT_MAKE_TARGET_mx6 = "u-boot.imx"
+WKS_FILE_mx6 = "imx-uboot-bootpart.wks"
```

Comment:
The problem in doing this, this way, is because this overrides values for other machines. It would be good if you could do that for the specific machines so it avoids breaking other people setups and allow the use of `fsl` distro for more people.

Reply:
Can I change the machine config settings to use fslc distro override?

UBOOT_SUFFIX_fslc = "img"
SPL_BINARY_fslc = "SPL"
UBOOT_MAKE_TARGET_fslc = ""
WKS_FILE_fslc = "imx-uboot-spl-bootpart.wks"

Reply:
I made https://github.com/Freescale/meta-freescale/pull/46 in reponse. It replaces this PR.

Reply:
No, the best way to do that is instead use a fsl override for this, on the distro layer, and add the override to the machine.

An example is:

https://github.com/UpdateHub/meta-updatehub-freescale/blob/master/conf/layer.conf#L19-L23

and

https://github.com/UpdateHub/meta-updatehub-freescale/blob/master/conf/layer.conf#L42-L47

Reply:
Okay, I'm having a hard time wrapping my head around all these overrides. Is the idea behind your suggestion that the default machine config is generally for fslc operation? I feel like this is not consistent, and see fslc overrides in the very example you send. Or is the idea that at this point you don't want to break compatibility?
What do I call this new override? machine-override-for-fsl-distro?

Reply:
No, the problem is meta-freescale should not depend on a distro, never. meta-freescale-distro is responsible of adding the distro and thus responsible for adding the overrides.

I'd use `fsl-bsp` as it in fact making the machine to use the `fsl-bsp` components and settings.

Reply:
Okay. Is there a consistency to the design? The default settings are correct for fsl, but then the machine config changes it to fslc. Now I will have to switch it back to fsl in the distro. Is the goal of the machine config to eventually be consistent on fslc config?

Reply:
Sure; we are based on U-Boot mainline and this is what causes most of the difference; once NXP upgrades to new U-Boot releases and finishes to upstreaming (which is very active by the way) it will converge.